### ISSUE TITLE
[Bug] LMB trigger buttons even when disabled

### ISSUE THREAD

**Issue by PhilJbt** [emvaized/circle-mouse-gestures#32]
When the option `Hide menu on mouse up if nothing selected` is `disabled` and `Open menu on` is set on `Right click`, **LMB** triggers buttons, but they do not work correctly. (v1.5)

I found this bug accidentally because it was more intuitive to me to use **RMB** to open the menu and **LMB** to select buttons. Be able to choose which mouse button between **RMB** and **LMB** triggers menu's buttons when the `Hide menu on mouse up if nothing selected` is `disabled` would be great.

In this example, the menu opening is set on **RMB**. Select the "*search*" button with **RMB** works, but with **LMB** the `{selection}` parameter is empty, resulting in a redirect to the Google homepage:

[2024-11-05 11-41-37.webm](https://github.com/user-attachments/assets/05edd493-3399-41c3-b5a8-8942b323d032)

**Comment by emvaized** [comment]
Thanks for your detailed report! 

Sure, it looks like an oversight that LMB doesn't proccess the selected action correctly. I'll take a look at it when I will have free time.

**Comment by emvaized** [comment]
Please let me know if the LMB works correctly in latest test build from [this comment](https://github.com/emvaized/circle-mouse-gestures/issues/33#issuecomment-2465817458)

**Comment by PhilJbt** [comment]
With the [test_2](https://github.com/emvaized/circle-mouse-gestures/issues/33#issuecomment-2465817458) build, when the menu is set up to **RMB**, click on a menu's button with **LMB** closes the menu without triggering anything.

**Comment by emvaized** [comment]
Is it with `Hide menu on mouse up if nothing selected` still disabled?

**Comment by PhilJbt** [comment]
`Hide menu on mouse up if nothing selected` was enabled.

- Disabled:
  - LMB = trigger, fail ❌
  - RMB = trigger, success ✔
  - Both = not triggered, close the menu ⚠

- Enabled:
  - Both = not triggered, close the menu ⚠
  - RMB = trigger, success ✔

It seems **RMB** has an `onPress` `triggerEvent` and **LMB** an `onRelease` ? This is a detail, but use both should not close the menu.  In this case, this could help you handle unexpected behavior by triggering a menu's button only if either **LMB** or **RMB** is pressed and then released. If at release, the other **MB** was previously registered as pressed, the menu should not disappear. But this may not help you at all in any way!

[2024-11-09 18-25-48.webm](https://github.com/user-attachments/assets/609b12ff-ee8e-4e50-a677-fca65955604b)

**Comment by emvaized** [comment]
> It seems **RMB** has an `onPress` `triggerEvent` and **LMB** an `onRelease`

Yes, in this build I added additional logic to mouse down event for the left mouse button. It probably collides with the existing logic to show the regular context menu on right click when left mouse button is held.

**Comment by emvaized** [comment]
I suddenly realised that your issues might be caused by interference with Vivaldi's own build-in mouse gestures, which are enabled by default: 

![illustration](https://github.com/user-attachments/assets/f43eaab6-36d4-4b89-955b-481c6a99d05d) 

Could you try disabling them and check if problems dissapeared?

**Comment by PhilJbt** [comment]
I disabled both Vivaldi's MG when I installed CMG. You can't imagine how fast things went wild!
But thank you; I appreciate your thoughtfulness.
